src/account_data.cairo

```
@@ -487,6 +487,43 @@ pub mod AccountData {
                 false
             }
         }
+        fn transaction_list(
+            self: @ComponentState<TContractState>, start: Option<u64>, limit: Option<u64>
+        ) -> Array<Transaction> {
+            let transaction_count = self.tx_count.read();
+
+            if transaction_count == 0 {
+                return ArrayTrait::new();
+            }
+
+            let start_idx: u256 = match start {
+                Option::Some(s) => s.into(),
+                Option::None => 0,
```

Looking at the implementation of how the transaction is created, the default should be 1 instead of 0 as we are using a one-index. Please update appropraitely.
Also, providing zero for the start and limit parameter should raise an error.